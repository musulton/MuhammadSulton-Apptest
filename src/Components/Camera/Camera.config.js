import Constants from '../../Constants';

const actions = [
  {
    title: 'Take Image',
    type: Constants.CAMERA_ACTION_TYPE.CAPTURE,
    options: {
      saveToPhotos: true,
      mediaType: 'photo',
      includeBase64: false,
    },
  },
  {
    title: 'Select Image',
    type: Constants.CAMERA_ACTION_TYPE.LIBRARY,
    options: {
      maxHeight: 200,
      maxWidth: 200,
      selectionLimit: 0,
      mediaType: 'photo',
      includeBase64: false,
    },
  },
];

export default {
  actions,
};
