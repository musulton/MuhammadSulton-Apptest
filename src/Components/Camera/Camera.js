import * as React from 'react';
import {View} from 'react-native';
import {launchCamera, launchImageLibrary} from 'react-native-image-picker';

import {Avatar, Button} from '../index';
import Constants from '../../Constants';

import styles from './Camera.styles';
import config from './Camera.config';

const Camera = () => {
  const [resourcePath, setResourcePath] = React.useState({
    assets: [],
  });

  const onButtonPress = React.useCallback((type, options) => {
    const imageMethod =
      type === Constants.CAMERA_ACTION_TYPE.CAPTURE
        ? launchCamera
        : launchImageLibrary;
    imageMethod(options, setResourcePath);
  }, []);

  return (
    <View style={styles.container}>
      <Avatar uri={resourcePath.assets[0]?.uri} size={150} fromGallery />
      <View style={styles.buttonContainer}>
        {config.actions.map(({title, type, options}, i) => {
          return (
            <Button
              key={i}
              text={title}
              onPress={() => onButtonPress(type, options)}>
              {title}
            </Button>
          );
        })}
      </View>
    </View>
  );
};

export default Camera;
