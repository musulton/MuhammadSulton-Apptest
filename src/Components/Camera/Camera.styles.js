import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: 160,
    marginVertical: 10,
  },
});
