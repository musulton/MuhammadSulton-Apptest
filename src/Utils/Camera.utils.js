import {Alert, PermissionsAndroid} from 'react-native';
import {launchCamera} from 'react-native-image-picker';
import Constants from '../Constants';

export const requestCamera = async (options, setResourcePath) => {
  try {
    const granted = await PermissionsAndroid.request(
      PermissionsAndroid.PERMISSIONS.CAMERA,
    );
    if (granted === PermissionsAndroid.RESULTS.GRANTED) {
      launchCamera(options, setResourcePath);
    } else {
      Alert.alert(
        'Info',
        "You haven't given the app permission to access the camera",
        [
          {
            text: 'Ok',
          },
        ],
      );
    }
  } catch (e) {
    console.warn(e);
  }
};
