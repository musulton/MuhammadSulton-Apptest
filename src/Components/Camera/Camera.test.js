import * as React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react-native';
import {launchImageLibrary} from 'react-native-image-picker';

import {requestCamera} from '../../Utils/Camera';

import Camera from './Camera';

jest
  .mock('../../Utils/Camera', () => ({
    requestCamera: jest.fn(),
  }))
  .mock('react-native-image-picker', () => ({
    launchImageLibrary: jest.fn(),
  }));

describe('Camera', () => {
  afterEach(cleanup);

  it('should render camera correctly', () => {
    const {toJSON} = render(<Camera />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should call requestCamera when Take Image button is clicked', () => {
    const {getByText} = render(<Camera />);

    const button = getByText('Take Image');
    fireEvent.press(button);

    expect(requestCamera).toBeCalled();
  });

  it('should call launchImageLibrary when Select Image button is clicked', () => {
    const {getByText} = render(<Camera />);

    const button = getByText('Select Image');
    fireEvent.press(button);

    expect(launchImageLibrary).toBeCalled();
  });
});
