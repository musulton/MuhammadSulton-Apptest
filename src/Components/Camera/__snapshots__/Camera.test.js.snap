// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera should render camera correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#9DE0AD",
        "borderRadius": 50,
        "height": 200,
        "justifyContent": "center",
        "width": 200,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï€‡
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 10,
        "width": 250,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Take Image
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Select Image
      </Text>
    </View>
  </View>
</View>
`;
